server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1200,7 +1200,8 @@ public void doInTransactionWithoutResult(TransactionStatus status) {
                     }
                 });
             } else {
-                throw new CloudRuntimeException("Cannot delete pool " + sPool.getName() + " as there are associated " + "non-destroyed vols for this pool");
+                logger.debug("Cannot delete storage pool {} as the following non-destroyed volumes are on it: {}.", sPool.getName(), getStoragePoolNonDestroyedVolumesLog(sPool.getId()));
```

Comment:
```suggestion
                logger.debug("Cannot delete storage pool {} as the following non-destroyed volumes are on it: {}.", sPool.getName(), () -> getStoragePoolNonDestroyedVolumesLog(sPool.getId()));
```

When executing heavy processes to load data only to log it, call the methods inside a lambda; this way, they will be evaluated only at the right log level.